src/app/app.component.ts

```diff
@@ -51,20 +51,23 @@ export class AppComponent implements OnInit, OnDestroy {
       this.podcasts = state;
 
       if (this.podcasts && this.podcasts.length > 0) {
-        const selectedPodcast = this.podcasts[0]; // Default select the first one. It'll be the last one that was updated
-        if (selectedPodcast &&
-          (!this.filter || !this.filter.podcast ||
-          this.filter.podcast && selectedPodcast.seriesId !== this.filter.podcast.seriesId)) {
-          this.store.dispatch(new CastleFilterAction({filter: {podcast: selectedPodcast}}));
+        if (!this.filterStoreSub) {
+          this.filterStoreSub = this.store.select(selectPodcastFilter).subscribe((newPodcastSeriesId: number) => {
+            if (newPodcastSeriesId && newPodcastSeriesId !== this.filteredPodcastSeriesId) {
+              const selectedPodcast = this.podcasts.find(p => p.seriesId === newPodcastSeriesId);
+              if (selectedPodcast) {
+                this.getEpisodes(selectedPodcast);
+              }
+            }
+            this.filteredPodcastSeriesId = newPodcastSeriesId;
+          });
         }
-      }
-    });
 
-    this.filterStoreSub = this.store.select(selectFilter).subscribe((newFilter: FilterModel) => {
-      if (newFilter.podcast && (!this.filter || !this.filter.podcast || newFilter.podcast.seriesId !== this.filter.podcast.seriesId)) {
-        this.getEpisodes(newFilter.podcast);
+        // Default select the first one by navigating to it. (It'll be the last one that was updated)
```

Comment:
instead of doing a filter action directly to set the selected podcast, navigate to the route and the routing effect will send a filter action to set the selected podcast